packages/frontend/src/redux/actions/account.js

```diff
@@ -43,7 +43,7 @@ export const getProfileStakingDetails = (accountId) => async (dispatch, getState
 
     const lockupIdExists = accountId
         ? !!getState().allAccounts[accountId].balance.lockedAmount
```

Comment:
So this check is for external accounts (if you use `/profile/:accountId`) only. I changed the arg name so it will be a bit more readable.

Reply:
Understood that the data is different, but why do we _not_ need the same fix for 'external accounts'? I would assume that we're going to be showing incorrect lockup account details for 'external' accounts unless we make a similar change to `allAccounts[accountId]` to track `hasLockup` under similar circumstances as the primary account.

Reply:
It's working properly, we don't need `hasLockup` because if we are showing an external account profile, this check:
`!!getState().allAccounts[externalAccountId].balance.lockedAmount`
is working properly and if there is no lockup the `hasLockup` is `undefined`.

Added `hasLockup` to the `allAccounts` reducer, it will be more consistent and more readable.

Generally, the problem is with handling `balance` object in `account` reducer (that was also causing the main problem in this issue) -  it's a bit complicated implementation, probably because there were a million changes to it, while the project was expanding by new functionalities. We should check the balance more deeply, probably create a separate reducer for it, and refactor it step by step - but it's a bit bigger work, if you prefer I can handle it now in separate PR, but the other things would have to wait a bit?